File: back/lms/src/main/java/com/lms/model/Book.java
package com.lms.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;
import com.fasterxml.jackson.annotation.JsonIgnore;
import java.util.*;
import java.time.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "book")
public class Book {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	@NotEmpty
	@Column(nullable = false)
	private String title;
	@Column
	private String isbn;
	@Column
	private LocalDate publishedDate;
	@Column
	private String language;
	@NotNull
	@Column(nullable = false)
	private Integer availableCopies;
	@NotNull
	@Column(nullable = false)
	private Integer totalCopies;
	@ManyToOne
	@JoinColumn(name = "author", referencedColumnName = "id")
	private Author author;

	@ManyToOne
	@JoinColumn(name = "publisher", referencedColumnName = "id")
	private Publisher publisher;

	@ManyToOne
	@JoinColumn(name = "category", referencedColumnName = "id")
	private Category category;

	@ManyToOne
	@JoinColumn(name = "library", referencedColumnName = "id")
	private Library library;


}

File: front/lms-frontend/src/app/features/entities/book/book.ts
import { Author } from '../author/author';
import { Publisher } from '../publisher/publisher';
import { Category } from '../category/category';
import { Library } from '../library/library';

export class Book {
	title!: string;
	isbn!: string;
	publishedDate!: Date;
	language!: string;
	availableCopies!: number;
	totalCopies!: number;
	author!: Author;
	publisher!: Publisher;
	category!: Category;
	library!: Library;
}
